### DISCUSSION TITLE
Scripting for UI and/or plugin routing

### DISCUSSION THREAD

**Discussion by zolkin** [FundamentalFrequency/.github#32] [Software]
Hi @stonepreston,

Thank you for your efforts with the project! The idea makes me incredibly inspired and excited;)

Quick question, do you consider adding scripting like Lua or JS/Typescript (no python, please) to the project to simplify programming efforts for something like UI or plugin routing, or maybe even some MIDI event processing?

I'm a C++ programmer and despite I have warm feelings for the language some programming efforts ask for faster development iteration cycles and internal scripting.

Interested in your thoughts on the subject,

Evgeny

**Comment by stonepreston** [comment]
I dont see any LUA/JS/TS/etc scripting happening unless its added upstream in JUCE itself. Having everything be written in the same language and the same framework minimizes complexity quite a bit, and I think the downsides of C++ are heavily outweighed by the benefits of a singular language encompassing the UI, DSP, etc. Things flow really well between JUCE and the tracktion engine, so it makes sense to stick with what they are using.
